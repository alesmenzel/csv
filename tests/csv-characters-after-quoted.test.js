const fs = require('fs')
const path = require('path')
const {pipeline} = require('stream')
const { promisify } = require('util')
const Parser = require('../csv-parser')
const Collect = require('./collect')

const pipelineAsync = promisify(pipeline)

it('CSV: relax characters after quoted cell', async () => {
  const parser = new Parser({ relaxCharactersAfterQuotedText: true })
  const collect = new Collect()
  const input = fs.createReadStream(path.resolve(__dirname, 'csv-characters-after-quoted.csv'))
  await pipelineAsync(input, parser, collect)
  expect(collect.getData()).toMatchSnapshot('csv-characters-after-quoted.csv')
})
