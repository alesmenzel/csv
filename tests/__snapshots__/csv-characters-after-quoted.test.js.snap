// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV: relax characters after quoted cell: csv-characters-after-quoted.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a",
    "\\"b",
    "c",
  ],
  Array [
    "x",
    "y",
    "z",
  ],
]
`;
